src/main/java/com/google/cloud/spanner/myadapter/statements/SessionStatementParser.java

```diff
@@ -0,0 +1,285 @@
+
+      Builder systemVariable() {
+        this.systemVariable = true;
+        return this;
+      }
+
+      Builder global() {
+        this.scope = SessionVariableScope.GLOBAL;
+        return this;
+      }
+
+      Builder name(String name) {
+        this.name = name;
+        return this;
+      }
+
+      Builder value(String value) {
+        this.value = value;
+        return this;
+      }
+
+      SetStatement build() {
+        return new SetStatement(this);
+      }
+    }
+
+    boolean systemVariable;
+    SessionVariableScope scope;
+    String name;
+    String value;
+
+    SetStatement(Builder builder) {
+      this.systemVariable = builder.systemVariable;
+      this.scope = builder.scope;
+      this.name = builder.name;
+      this.value = builder.value;
+    }
+
+    @Override
+    public StatementResult execute(SessionState sessionState) {
+      sessionState.set(name, value, scope);
+      return new UpdateCount(0L);
+    }
+  }
+
+  static class SelectStatement extends SessionStatement {
+    static class Builder {
+      Builder addVariableColumn(VariableColumn variableColumn) {
+        variableColumnList.add(variableColumn);
+        return this;
+      }
+
+      ArrayList<VariableColumn> variableColumnList = new ArrayList<VariableColumn>();
+    }
+
+    SelectStatement(Builder builder) {
+      this.variableColumns = builder.variableColumnList;
+    }
+
+    @Override
+    public StatementResult execute(SessionState sessionState) {
+      ArrayList<StructField> structFields = new ArrayList<Type.StructField>();
+      Struct.Builder rowBuilder = Struct.newBuilder();
+      for (VariableColumn column : variableColumns) {
+        SystemVariable variable = sessionState.get(column.variableName, column.scope);
+        structFields.add(StructField.of(column.columnName, variable.getType()));
+        rowBuilder
+            .set(column.columnName)
+            .to(Parser.parseToSpannerValue(variable.getValue(), variable.getType().getCode()));
+      }
+
+      Type rowType = Type.struct(structFields);
+      ImmutableList<Struct> rows = ImmutableList.of(rowBuilder.build());
+      Preconditions.checkNotNull(rowType);
+      Preconditions.checkNotNull(rows);
+      Preconditions.checkArgument(rowType.getCode() == Code.STRUCT);
+      ResultSet resultSet = ResultSets.forRows(rowType, rows);
+      return new QueryResult(resultSet);
+    }
+
+    ArrayList<VariableColumn> variableColumns;
+  }
+
+  public static @Nullable SessionStatement parse(ParsedStatement parsedStatement) {
+    if (parsedStatement.getType() == StatementType.CLIENT_SIDE) {
+      // TODO : See if we need special handling for client side statements.
+      // Disabling this right now as "SET AUTOCOMMIT 1" is getting treated as
+      // client side statement, but it's not a valid GoogleSQL statement.
+    }
+    SimpleParser parser = new SimpleParser(parsedStatement.getSqlWithoutComments());
+    if (parser.eatKeyword("set")) {
```

Comment:
Validate is SET also works. Also use static variables.

Reply:
Yes SimpleParser methods like eatKeyWord and eatToken ignores the case. So both SET and set works. Here as well as other places also.